libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1967,7 +1979,15 @@ class MapboxNavigation @VisibleForTesting internal constructor(
             internalSetNavigationRoutes(
                 routes,
                 SetRoutes.Reroute
-            )
+            ) {
+                it.onError {
+                    logE(LOG_CATEGORY) {
+                        "Failed to set route generated by the reroute controller, " +
+                            "reason: ${it.message}. Retrying."
+                    }
+                    reroute()
```

Comment:
Do we expect to get a new (better) route here? Why would it change?
I'm worried that we will be forever rerouting in a loop if something goes wrong. Can we set the maximum number of retries?

Reply:
> Do we expect to get a new (better) route here? Why would it change?

This is in case Nav Native fails to accept the route, so possibly a somewhat malformed or outdated response from the backend.

I'll cut a separate PR with this.